src/appleseed/foundation/meta/tests/test_aabb.cpp

```diff
@@ -387,6 +426,21 @@ TEST_SUITE(Foundation_Math_AABB)
         EXPECT_FEQ(Vector3d(5.0, 7.0, 9.0), bbox.extent());
     }
 
+    TEST_CASE(TestExtentOnIntegralAABB)
+    {
+        const AABB3i bbox(
+            Vector3i(-1, -2, -3),
+            Vector3i(4, 5, 6));
+
+        EXPECT_FEQ(Vector3i(6, 8, 10), bbox.extent());
+
+        const AABB3i bbox2(
+            Vector3i(0, -1, 1),
+            Vector3i(4, 4, 4));
+
+        EXPECT_FEQ(Vector3i(5, 6, 4), bbox2.extent());
```

Comment:
One test is enough, you can remove this one unless it's testing another corner case in which case it's best to make it a separate test with an explicit name.